sphinxcontrib/needs/directives/needtable.py

```diff
@@ -154,13 +166,19 @@ def process_needtables(app, doctree, fromdocname):
         tgroup = nodes.tgroup()
 
         # Define Table column width
-        # ToDo: Find a way to chosen to perfect width automatically.
+        colwidths_counter = 0
+        colwidths = current_needtable["colwidths"]
         for option, title in current_needtable["columns"]:
-            if option == "TITLE":
+
+            if colwidths:  # Get values from given colwidths option
+                tgroup += nodes.colspec(colwidth=int(colwidths[colwidths_counter]))
```

Comment:
you already coerced the elements of `colwidths` to `int` above (line 65)

Reply:
You are right, removed.